test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJvmtiAPI.java

```diff
@@ -17,24 +17,39 @@
     public static void main(String args[]) {
     	boolean ret = true;
     	String dir = null;
+    	String dirGroupAccess = null;
+    	String dirRemoveJavaSharedResources = null;
     	int oldCacheCount = 0;
+    	int oldCacheGroupAccessCount = 0;
     	int newCacheCount = 0;
+    	int newCacheGroupAccessCount = 0;
     	
         runDestroyAllCaches();
         if (false == isWindows()) {
         	runDestroyAllSnapshots();
+        	if (isOpenJ9()) {
+            	runDestroyAllGroupAccessCaches();
+            }
         }
-        try {
+       try {
 	    	dir = getCacheDir();
 	    	if (dir == null) {
+	    		dir = getControlDir();
 	    	}
-	    	
 	    	/* get cache count before creating any new cache */
 	    	oldCacheCount = iterateSharedCache(dir, NO_FLAGS, false);
 		    if (oldCacheCount == -1) {
 		    	fail("iterateSharedCacheFunction failed");
 		    }
-		    	    
+	    	
+	    	if (dir == null && false == isWindows() && false == isMVS() && isOpenJ9()) {
+	    		dirGroupAccess = getCacheDir("Foo_groupaccess", false);
+	    		dirRemoveJavaSharedResources = removeJavaSharedResourcesDir(dirGroupAccess);
+	    		oldCacheGroupAccessCount = iterateSharedCache(dirGroupAccess, NO_FLAGS, false) + iterateSharedCache(dirRemoveJavaSharedResources, NO_FLAGS, false);
+			    if (oldCacheGroupAccessCount == -1) {
+			    	fail("iterateSharedCacheFunction failed");
```

Comment:
Better to include the directory in the error message

Reply:
Updated.